roles.cpp

```diff
@@ -0,0 +1,163 @@
+////////////////////////////////////////////////////
+// Automatically generated by "generate-roles.go" //
+////////////////////////////////////////////////////
+
+#include "roles.h"
+
+static const char *no_rule = "role.empty";
+static const char *unknown_rule = "role.unknown";
+static const char *id_to_roles[] = {
+    "role.Role(0)",
```

Comment:
yeah! I noticed that, but "role_name_for_id" never returns "role.Role(0)".

it is one artifact of using an array and not having a defined name for the value 0.